tests/CMakeLists.txt

```diff
@@ -50,3 +50,51 @@ target_link_libraries(tests ${APPD_SDK_LINK_LIBS} ${CGREEN_LINK_LIBS})
 add_custom_target(run-tests COMMAND ./tests)
 
 add_dependencies(run-tests tests appdynamicsiotsdk)
+
+##########################################
+# Target
+# run-code-coverage : create a code coverage report
+##########################################
+
+if (ENABLE_COVERAGE)
+    FIND_PROGRAM(GCOV_PATH gcov)
+    FIND_PROGRAM(LCOV_PATH lcov)
+    FIND_PROGRAM(GENHTML_PATH genhtml)
+
+    IF(NOT GCOV_PATH)
+        MESSAGE(FATAL_ERROR "gcov not found! Aborting...")
+    ENDIF() # NOT GCOV_PATH
+
+    IF(NOT LCOV_PATH)
+        MESSAGE(FATAL_ERROR "lcov not found! Aborting...")
+    ENDIF() # NOT LCOV_PATH
+
+    IF(NOT GENHTML_PATH)
+        MESSAGE(FATAL_ERROR "genhtml not found! Aborting...")
+    ENDIF() # NOT GENHTML_PATH
+endif()
```

Comment:
This endif() should move to the end of file to include gcov, lcov commands.

Reply:
The first block causes the cmake itself to abort, and all `make` commands will fail.

Reply:
Done